### PR TITLE
Added the get, set props instead of hardcoded values and directly converting the aprapoint3f to vtkDataArray

### PR DESCRIPTION
**Pull request by K-aniket23**
Title: Enhance vtkIntegartion Module: Data Handling and Property Management

Description:
This pull request enhances the VTK module by improving data handling efficiency and flexibility:    

1: Employed VTK functions to copy data from aprapoints3f to a vtkArray, eliminating the need to copy data from aprapoints3f to vtkPoints using a for loop.

2: Added getProps and setProps functions to replace hardcoded values for properties, enhancing flexibility.

3: Implemented unit tests to evaluate the functionality of the newly added get and setProps functions.

### CODE REVIEW

File: base/src/STLRendererSink.cpp [hunk 1]
```diff
@@ -90,10 +93,17 @@ class Detail
 		iren->TerminateApp();
 	}
 
+	void setProps(STLRendererSinkProps &_props)
+	{
+		props = _props;
+	}
+
+public:
+	STLRendererSinkProps props;
```
Comment (abhi9sh) [inline base/src/STLRendererSink.cpp:102]: declare this props at top

**Review by abhi9sh — CHANGES_REQUESTED** [review]

File: base/test/stlrenderer_tests.cpp [hunk 2]
```diff
@@ -7,4407 +7,8733 @@
...
+		{0.07641913, -0.10807946, 0.08696543},
+		{0.091404535, -0.0921432, 0.079539225},
+		{0.08954233, -0.092671275, 0.09013894},
+		{0.07641913, -0.10807946, 0.08696543},
+		{0.08954233, -0.092671275, 0.09013894},
+		{0.08334184, -0.09433231, 0.09526661},
+		{0.07641913, -0.10807946, 0.08696543},
+		{0.08334184, -0.09433231, 0.09526661},
+		{0.071751535, -0.1027972, 0.09746154},
+		{0.07641913, -0.10807946, 0.08696543},
```
Comment (abhi9sh) [inline base/test/stlrenderer_tests.cpp:8661]: use sensible variable name

**Review by abhi9sh — CHANGES_REQUESTED** [review]

File: base/include/STLRendererSink.h [hunk 3]
```diff
@@ -15,61 +15,78 @@
...
+	{
+		meshDiffuseColor = _meshDiffuseColor;
+		meshSpecularCoefficient = _meshSpecularCoefficient;
+		meshSpecularPower = _meshSpecularPower;
+		cameraPosition = _cameraPosition;
+		cameraFocalPoint = _cameraFocalPoint;
+		winName = _winName;
+		winWidth = _winWidth;
+		winHeight = _winHeight;
+		bkgColor = _bkgColor;
```
[the remark this answers is not in the document]
Comment (K-aniket23) [reply base/include/STLRendererSink.h:33]: should i do this as it would not be more clean and clear code

**Review by mraduldubey — APPROVED** [review]

File: base/test/stlrenderer_tests.cpp [hunk 4]
```diff
@@ -7,4407 +7,8733 @@
...
-{-0.007829636, -0.0047226623, 0.13480772},
-{0.00077722594, -0.0029412694, 0.13448451},
-{-0.007829636, -0.0047226623, 0.13480772},
-{-0.0065494142, -0.012701452, 0.1330847},
-{0.00077722594, -0.0029412694, 0.13448451},
-{-0.0065494142, -0.012701452, 0.1330847},
-{0.0013935305, -0.011261113, 0.1330998} };
+public:
+	SourceModule(SourceModuleProps props = SourceModuleProps()) : Module(SOURCE, "SourceModule", props)
+	{
```
[the remark this answers is not in the document]
Comment (mraduldubey) [reply base/test/stlrenderer_tests.cpp:31]: yes

[the remark this answers is not in the document]
Comment (mraduldubey) [reply base/test/stlrenderer_tests.cpp:31]: Ideally, should be read from file. Best way is to use vtk itself to read and iterate over the vtkarray to give  the points when source module does step ()

**Review by mraduldubey — CHANGES_REQUESTED** [review]

**Pull request merged**